skfda/representation/basis.py

```diff
@@ -2175,6 +2239,13 @@ def inner_product(self, other, lfd_self=None, lfd_other=None,
         if weights is not None:
             other = other.times(weights)
 
+        if self.nsamples * other.nsamples > self.nbasis * other.nbasis:
+            return self.coefficients @ self.basis.inner_matrix(other.basis) @ other.coefficients.T
+        else:
+            return self._inner_product_integrate(other, lfd_self, lfd_other)
+
+    def _inner_product_integrate(self, other, lfd_self, lfd_other):
```

Comment:
Is this still necessary?

Reply:
Yes, because if you want to calculate the inner product between two objects with 1 sample each with 10 basis each, it computes only one integral instead of the 100 integrals it has to make for the inner matrix and then the matrix product. I filter the method base on the number of integrals it will do
```python
if self.nsamples * other.nsamples > self.nbasis * other.nbasis:
    inner matrix
else:
    numerical integration
```

This is the check for computing the minimun number of integrals

Reply:
Seems reasonable.